renku/core/dataset/providers/local.py

```diff
@@ -114,14 +115,30 @@ def get_metadata(
         if flags > 1:
             raise errors.ParameterError("--move, --copy and --link are mutually exclusive.")
 
-        prompt_action = True if flags == 0 else False
+        prompt_action = False
 
         if move:
             default_action = DatasetAddAction.MOVE
         elif link:
             default_action = DatasetAddAction.SYMLINK
-        else:
+        elif copy:
             default_action = DatasetAddAction.COPY
+        else:
+            prompt_action = True
+            action = get_value("renku", "default_dataset_add_action")
+            if action:
+                prompt_action = False
+                if action.lower() == "copy":
+                    default_action = DatasetAddAction.COPY
+                elif action.lower() == "move":
+                    default_action = DatasetAddAction.MOVE
+                else:
```

Comment:
should we allow setting `link` as a default as well, but prompt with `--copy` for external files? I think that's nicer from a UX perspective, if users can set all three actions as a default.

But U wonder if would could split it, so LocalProvider only handles paths within a repo and ExternalProvider is used for files outside of a project? That might be cleaner.

Reply:
A default value of `link` for external files will prompt users what to do.

Reply:
> We don't allow linking for external files anymore. We have a whole dataset as external and we either mount it or symlink its data directory (not individual files).

Yes but I wonder if we should always use the External provider even for adding individual files (right now ExternalProvider is not an AddProvider). That would maybe help solve some of the confusion that could happen here